app/controllers/search_controller.rb

```diff
@@ -0,0 +1,15 @@
+class SearchController < ApplicationController
+  def results
+    return unless query.present?
+
+    search_service = ProjectSearchService.new
+    @results = search_service.search(query)
+    @query = query
+    @count = @results.count
+    @pager, @results = pagy_array(@results)
+  end
+
+  private def query
+    params[:query]
```

Comment:
I don't know if it should be here, but we should `strip` the query string, trim leading and trailing white space.